script/production/ConfigWhitelistOracleAdapters.sh

```diff
@@ -0,0 +1,97 @@
+
+    if [[ "$adapter" == "Adapter" ]]; then
+        continue
+    fi
+
+    entry=$(cast call $adapter_registry "entries(address)((uint128,uint128))" $adapter --rpc-url $DEPLOYMENT_RPC_URL)
+    addedAt=$(echo $entry | cut -d'(' -f2 | cut -d',' -f1)
+    revokedAt=$(echo $entry | cut -d',' -f2 | cut -d')' -f1 | tr -d ' ')
+
+    if [[ "$whitelist" == "Yes" ]]; then
+        if [[ "$base" == "" || "$quote" == "" ]]; then
+            echo "No base or quote address found for the adapter $adapter. Skipping..."
+            continue
+        fi
+
+        if [[ $addedAt == "0" && $revokedAt == "0" ]]; then
+            if [ "$is_batch" != "y" ]; then
+                gasPrice=$(cast gas-price --rpc-url "$DEPLOYMENT_RPC_URL")
+
+                if cast send $adapter_registry "add(address,address,address)()" $adapter $base $quote --rpc-url $DEPLOYMENT_RPC_URL --private-key $DEPLOYER_KEY --legacy --gas-price $gasPrice > /dev/null; then
+                    echo "Successfully added adapter $adapterName ($adapter) to the registry."
+                else
+                    echo "Failed to add adapter $adapterName ($adapter) to the registry."
+                fi
+            else
+                echo "Adding 'add' batch item for adapter $adapterName ($adapter)."
+                items+="($adapter_registry,$onBehalfOf,0,$(cast calldata "add(address,address,address)" $adapter $base $quote)),"
+            fi
+        else
+            echo "Adapter $adapterName ($adapter) is already added to the registry or has been revoked. Skipping..."
+        fi
+    elif [[ "$whitelist" == "No" ]]; then
+        if [[ $addedAt != "0" && $revokedAt == "0" ]]; then
+            if [ "$is_batch" != "y" ]; then
+                gasPrice=$(cast gas-price --rpc-url "$DEPLOYMENT_RPC_URL")
+
+                if cast send $adapter_registry "revoke(address)()" $adapter --rpc-url $DEPLOYMENT_RPC_URL --private-key $DEPLOYER_KEY --legacy --gas-price $gasPrice > /dev/null; then
+                    echo "Successfully revoked adapter $adapterName ($adapter) from the registry."
+                else
+                    echo "Failed to revoke adapter $adapterName ($adapter) from the registry."
+                fi
+            else
+                echo "Adding 'revoke' batch item for adapter $adapterName ($adapter)."
+                items+="($adapter_registry,$onBehalfOf,0,$(cast calldata "revoke(address)" $adapter)),"
+            fi
+        else
+            echo "Adapter $adapterName ($adapter) is not added yet to the registry or has been revoked. Skipping..."
+        fi
+    else
+        echo "Invalid Whitelist value for adapter $adapterName ($adapter). Skipping..."
+    fi
+done < <(tr -d '\r' < "$csv_file")
+
+items="${items%,}]"
+
+echo "Executing batch transaction..."
+if [ "$is_batch" == "y" ]; then
+    if cast send $evc "batch((address,address,uint256,bytes)[])" $items --rpc-url $DEPLOYMENT_RPC_URL --private-key $DEPLOYER_KEY --legacy > /dev/null; then
```

Comment:
I agree with getting rid of the `--gas-price` arg altogether and letting the user specify with `ETH_GAS_PRICE` env variable. Dynamically computing it makes more sense if you're sending a bunch of txns so it can adjust over time, but here we're just sending one big batch.